samples/js/flickr/find-photos/src/main.js

```diff
@@ -76,10 +77,14 @@ main().catch(ex => {
 });
 
 async function main(): Promise<void> {
-  const inSpec = DatasetSpec.parse(args._[0]);
-  const [db, input] = await inSpec.value();
+  const inSpec = PathSpec.parse(args._[0]);
+  const pinnedSpec = await inSpec.pin();
+  if (!pinnedSpec) {
+    throw `Input dataset ${inSpec.path.dataset} does not exist`;
+  }
+  const [db, input] = await pinnedSpec.value();
   if (!input) {
-    return db.close();
+    throw `Input spec ${args._[0]} does not exist`;
```

Comment:
Because it causes the error handling above to not print the stack. Kinda hack, but works out nicely. We use this in other js programs.